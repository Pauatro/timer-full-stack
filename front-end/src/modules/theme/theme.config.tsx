import { EThemes, EThemeVariables } from "./theme.interfaces";

export const themeColors: {
    [key in EThemes]: {[key in EThemeVariables]: string}
} = {
    [EThemes.LIGHT]:{
        [EThemeVariables.BACKGROUNDPC]: "#f7f6f2",
        [EThemeVariables.BUTTONPC]: "#373738",
        [EThemeVariables.FONTPC]: "#373738",
        [EThemeVariables.FONTSC]: "#f7f6f2",
        [EThemeVariables.ICONSPC]: "#373738",   
        [EThemeVariables.ICONSSC]: "#f7f6f2",
        [EThemeVariables.SWITCHPC]: "#f7f6f2",
        [EThemeVariables.SWITCHSC]: "#373738",
    },
    [EThemes.DARK]:{
        [EThemeVariables.BACKGROUNDPC]: "#383838",
        [EThemeVariables.BUTTONPC]: "#f5f4f0",
        [EThemeVariables.FONTPC]: "#f5f4f0",
        [EThemeVariables.FONTSC]: "#383838",
        [EThemeVariables.ICONSPC]: "#f5f4f0",
        [EThemeVariables.ICONSSC]: "#383838",
        [EThemeVariables.SWITCHPC]: "#ffffff",
        [EThemeVariables.SWITCHSC]: "#828282",
    },
}