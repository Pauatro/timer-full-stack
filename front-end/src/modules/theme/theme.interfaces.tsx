
export enum EThemes{
    DARK = "dark",
    LIGHT = "light"
}

export enum EThemeVariables{
    BACKGROUNDPC = "background-primary-color",
    BUTTONPC = "button-primary-color",
    FONTPC = "font-primary-color",
    FONTSC = "font-secondary-color",
    ICONSPC = "icons-primary-color",
    ICONSSC = "icons-secondary-color",
    SWITCHPC = "switch-primary-color",
    SWITCHSC = "switch-secondary-color",
}