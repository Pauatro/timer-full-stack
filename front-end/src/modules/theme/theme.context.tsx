import React, { useState } from "react";
import { EThemes, EThemeVariables } from "./theme.interfaces";
import { themeColors } from "./theme.config"

interface IThemeContext {
    theme: EThemes;
    switchTheme: ()=>void;
}

const ThemeContext = React.createContext<IThemeContext>({
    theme: EThemes.LIGHT,
    switchTheme: ()=>{},
});

const ThemeProvider = (props: any) => {

    const [ theme, setTheme ] = useState<EThemes>(EThemes.LIGHT)

    const switchTheme = ()=>{
        const newTheme = theme === EThemes.LIGHT?EThemes.DARK:EThemes.LIGHT
        setTheme(newTheme);

        const variables = Object.keys(themeColors[newTheme]) as EThemeVariables[]

        variables.forEach((variable: EThemeVariables)=>{
            document.documentElement.style.setProperty(`--${variable}`, themeColors[newTheme][variable])
        })
    }

    return (
        <ThemeContext.Provider
            value={{
                theme,
                switchTheme
            }}
        >
            {props.children}
        </ThemeContext.Provider>
    );
};

export { ThemeProvider, ThemeContext };