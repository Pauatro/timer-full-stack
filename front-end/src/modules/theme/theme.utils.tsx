import { EThemes } from "./theme.interfaces";

export const themeColors = {
    [EThemes.LIGHT]:{
        background: "#f7f6f2",
        buttons: "#373738",
        fontPrimary: "#373738",
        fontSecondary: "#f7f6f2",
        iconsPrimary: "#373738",
        iconsSecondary: "#f7f6f2",
        switchPrimary: "#f7f6f2",
        switchSecondary: "#373738",
    },
    [EThemes.DARK]:{
        background: "#383838",
        buttons: "#f5f4f0",
        fontPrimary: "#f5f4f0",
        fontSecondary: "#383838",
        iconsPrimary: "#f5f4f0",
        iconsSecondary: "#383838",
        switchPrimary: "#ffffff",
        switchSecondary: "#828282",
    },
}