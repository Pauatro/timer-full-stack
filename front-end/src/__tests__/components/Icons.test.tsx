import React from 'react';
import renderer from 'react-test-renderer';
import { 
    Icon,
    HeaderLogo,
    SunIcon,
    MoonIcon,
    PlayIcon,
    StopIcon
} from '../../components/Icons';
import { EThemes } from '../../modules/theme/theme.interfaces';

it('Icon renders correctly', () => {
  const tree = renderer
    .create(<Icon
        class = "icon"
        icon = "resources/logo-light.svg"
        alt = "icon"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('HeaderLogo renders correctly on LIGHT theme', () => {
    const tree = renderer
      .create(<HeaderLogo
          theme = {EThemes.LIGHT}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('HeaderLogo renders correctly on DARK theme', () => {
    const tree = renderer
      .create(<HeaderLogo
          theme = {EThemes.DARK}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('SunIcon renders correctly on LIGHT theme', () => {
    const tree = renderer
      .create(<SunIcon
          theme = {EThemes.LIGHT}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('SunIcon renders correctly on DARK theme', () => {
    const tree = renderer
      .create(<SunIcon
          theme = {EThemes.DARK}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('MoonIcon renders correctly on LIGHT theme', () => {
    const tree = renderer
      .create(<MoonIcon
          theme = {EThemes.LIGHT}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('MoonIcon renders correctly on DARK theme', () => {
    const tree = renderer
      .create(<MoonIcon
          theme = {EThemes.DARK}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('PlayIcon renders correctly on LIGHT theme', () => {
    const tree = renderer
      .create(<PlayIcon
          theme = {EThemes.LIGHT}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('PlayIcon renders correctly on DARK theme', () => {
    const tree = renderer
      .create(<PlayIcon
          theme = {EThemes.DARK}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('StopIcon renders correctly on LIGHT theme', () => {
    const tree = renderer
      .create(<StopIcon
          theme = {EThemes.LIGHT}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('StopIcon renders correctly on DARK theme', () => {
    const tree = renderer
      .create(<StopIcon
          theme = {EThemes.DARK}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});