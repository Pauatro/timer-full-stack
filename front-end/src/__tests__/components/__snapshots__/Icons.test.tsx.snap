// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLogo renders correctly on DARK theme 1`] = `
<img
  alt="MVST"
  className="header-logo"
  src="logo-dark.svg"
/>
`;

exports[`HeaderLogo renders correctly on LIGHT theme 1`] = `
<img
  alt="MVST"
  className="header-logo"
  src="logo-light.svg"
/>
`;

exports[`Icon renders correctly 1`] = `
<img
  alt="icon"
  className="icon"
  src="resources/logo-light.svg"
/>
`;

exports[`MoonIcon renders correctly on DARK theme 1`] = `
<img
  alt="Moon"
  className="moon-icon"
  src="moon-dark.svg"
/>
`;

exports[`MoonIcon renders correctly on LIGHT theme 1`] = `
<img
  alt="Moon"
  className="moon-icon"
  src="moon-light.svg"
/>
`;

exports[`PlayIcon renders correctly on DARK theme 1`] = `
<img
  alt="Start"
  className="play-icon"
  src="play-icon-dark.svg"
/>
`;

exports[`PlayIcon renders correctly on LIGHT theme 1`] = `
<img
  alt="Start"
  className="play-icon"
  src="play-icon-light.svg"
/>
`;

exports[`StopIcon renders correctly on DARK theme 1`] = `
<img
  alt="Stop"
  className="stop-icon"
  src="stop-icon-dark.svg"
/>
`;

exports[`StopIcon renders correctly on LIGHT theme 1`] = `
<img
  alt="Stop"
  className="stop-icon"
  src="stop-icon-light.svg"
/>
`;

exports[`SunIcon renders correctly on DARK theme 1`] = `
<img
  alt="Sun"
  className="sun-icon"
  src="sun-dark.svg"
/>
`;

exports[`SunIcon renders correctly on LIGHT theme 1`] = `
<img
  alt="Sun"
  className="sun-icon"
  src="sun-light.svg"
/>
`;
